src/meta/server_state.cpp

```diff
@@ -3269,8 +3269,373 @@ void server_state::set_max_replica_count(configuration_set_max_replica_count_rpc
+                    return;
+                }
+
+                auto &context = app->helpers->contexts[partition_index];
+                context.pending_sync_task =
+                    update_partition_max_replica_count_on_remote(new_partition_config, rpc);
+            },
+            server_state::sStateHash,
+            std::chrono::seconds(1));
+    }
+
+    ddebug_f("request for updating partition-level max_replica_count on remote storage: "
+             "app_name={}, app_id={}, partition_id={}, new_max_replica_count={}, new_ballot={}",
+             app_name,
+             app_id,
+             partition_index,
+             new_max_replica_count,
+             new_ballot);
+
+    auto partition_path = get_partition_path(gpid);
+    auto json_config =
+        dsn::json::json_forwarder<partition_configuration>::encode(new_partition_config);
+    return _meta_svc->get_remote_storage()->set_data(
+        partition_path,
+        json_config,
+        LPC_META_STATE_HIGH,
+        std::bind(&server_state::on_update_partition_max_replica_count_on_remote_reply,
+                  this,
+                  std::placeholders::_1,
+                  new_partition_config,
+                  rpc),
+        tracker());
+}
+
+// ThreadPool: THREAD_POOL_META_STATE
+void server_state::on_update_partition_max_replica_count_on_remote_reply(
+    error_code ec,
+    const partition_configuration &new_partition_config,
+    configuration_set_max_replica_count_rpc rpc)
+{
+    const auto &gpid = new_partition_config.pid;
+    const auto app_id = gpid.get_app_id();
+    const auto partition_index = gpid.get_partition_index();
+    const auto new_max_replica_count = new_partition_config.max_replica_count;
+    const auto new_ballot = new_partition_config.ballot;
+
+    const auto &app_name = rpc.request().app_name;
+    auto &response = rpc.response();
+
+    ddebug_f("reply for updating partition-level max_replica_count on remote storage: "
+             "error_code={}, app_name={}, app_id={}, partition_id={}, new_max_replica_count={}, "
+             "new_ballot={}",
+             ec.to_string(),
+             app_name,
+             app_id,
+             partition_index,
+             new_max_replica_count,
+             new_ballot);
+
+    zauto_write_lock l(_lock);
+
+    auto app = get_app_and_check_exist(app_name, response);
+    if (app == nullptr) {
+        response.hint_message +=
+            fmt::format(" while replying for updating max_replica_count of partition {} on remote",
+                        partition_index);
+        derror_f("failed to set max_replica_count: app_name={}, app_id={}, error_code={}, "
+                 "hint_message={}",
+                 app_name,
+                 app_id,
+                 response.err.to_string(),
+                 response.hint_message);
+        return;
+    }
+
+    // if multiple threads exist in the thread pool, the check may failed
+    dassert_f(
```

Comment:
same, however if the dassert for `transaction` , that is to say, if all previous process need drop via core-dump when persist remote failed, here may be necessary